loleaflet/src/layer/tile/CanvasTileLayer.js

```diff
@@ -6023,7 +6155,46 @@ L.CanvasTileLayer = L.Layer.extend({
 		}
 	},
 
+	_onTileMsgFileBasedView: function (textMsg, img) {
+		var tileMsgObj = app.socket.parseServerCmd(textMsg);
+		var ratio = [this._tileSize / this._tileWidthTwips, this._tileSize / this._tileHeightTwips];
+		var coords = new L.TileCoordData(tileMsgObj.x * ratio[0], tileMsgObj.y * ratio[1]);
+		coords.z = tileMsgObj.zoom;
+		coords.part = tileMsgObj.part;
+
+		var tile = this._tiles[coords.key()];
+
+		if (tile) {
+			if (tile._invalidCount > 0) {
+				tile._invalidCount -= 1;
+			}
+
+			tile.el.src = img;
```

Comment:
The above comment about slurping changes applies here too. (img is the resolved image object not the source string).